src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -1165,6 +1170,88 @@ void CascadiaSettings::CurrentDefaultTerminal(const Model::DefaultTerminal& term
     _currentDefaultTerminal = terminal;
 }
 
+winrt::hstring CascadiaSettings::AIEndpoint() noexcept

```

Comment:
Are all these things that should be done synchronously, on the UI thread? `TerminalPage` does call these in the `_extensionPalette.AIKeyAndEndpointRequested` handler... which happens in the visibility changed handler on the UI thread